File: Week-3/6.py
# Read matrix size
rows, cols = map(int, input("Enter rows and columns: ").split())

# Read matrix elements
matrix = []
print("Enter matrix elements row-wise:")
for i in range(rows):
    row = list(map(int, input().split()))
    matrix.append(row)

# Read rows to interchange
r1, r2 = map(int, input("Enter two row indices to interchange (0-based): ").split())

# Swap rows
matrix[r1], matrix[r2] = matrix[r2], matrix[r1]

print("Matrix after interchanging rows:")
for row in matrix:
    print(row)

File: Week-3/6.c
#include<stdio.h>
int main(){
    int rows,cols;
    printf("Enter number of rows and columns: ");
    scanf("%d %d", &rows, &cols);

    int matrix[rows][cols];
    printf("Enter matrix elements:\n");
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            scanf("%d", &matrix[i][j]);
        }
    }
    int r1,r2;
    printf("Enter two row indices to interchange: ");
    scanf("%d %d",&r1,&r2);
    for(int j = 0;j<cols;j++){
        int temp = matrix[r1][cols];
        matrix[r1][cols] = matrix[r2][cols];
        matrix[r2][cols] = temp;
    }
    printf("Matrix after interchanging rows:\n");
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
    return 0;
}